sdk/src/Core/Amazon.Util/EC2InstanceMetadata.cs

```diff
@@ -735,16 +686,20 @@ private static List<string> GetItems(string relativeOrAbsolutePath, int tries, b
             //token cannot be obtained we will fallback to not using a token.
             if (token == null)
             {
-                token = FetchApiToken(DEFAULT_RETRIES);
+                try
+                {
+                    token = FetchApiToken(DEFAULT_RETRIES);
+                }
+                catch(InvalidOperationException e)
+                {
+                    Logger.GetLogger(typeof(EC2InstanceMetadata)).InfoFormat("Failed to retrieve IMDS data \"{0}\" because IMDS API token could not be retrieved: {1}", relativeOrAbsolutePath, e.Message);
+                    return null; // If we could not get a token then assume we are not running in an EC2 instance and return null.
```

Comment:
Return null here is to maintain consistency with V3 that if the SDK failed to access IMDS for one of the properties of `EC2InstanceMetadata` null is returned to the user.